homeassistant/components/overkiz/climate_entities/hitachi_air_to_air_heat_pump_ovp.py

```diff
@@ -0,0 +1,351 @@
+    OverkizCommandParam.SILENT: FAN_SILENT,
+}
+
+FAN_MODES_TO_OVERKIZ: dict[str, str] = {
+    FAN_AUTO: OverkizCommandParam.AUTO,
+    FAN_HIGH: OverkizCommandParam.HI,
+    FAN_LOW: OverkizCommandParam.LO,
+    FAN_MEDIUM: OverkizCommandParam.MED,
+    FAN_SILENT: OverkizCommandParam.SILENT,
+}
+
+
+class HitachiAirToAirHeatPumpOVP(OverkizEntity, ClimateEntity):
+    """Representation of Hitachi Air To Air HeatPump."""
+
+    _attr_hvac_modes = [*HVAC_MODES_TO_OVERKIZ]
+    _attr_preset_modes = [PRESET_NONE, PRESET_HOLIDAY_MODE]
+    _attr_swing_modes = [*SWING_MODES_TO_OVERKIZ]
+    _attr_target_temperature_step = 1.0
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_translation_key = DOMAIN
+
+    def __init__(
+        self, device_url: str, coordinator: OverkizDataUpdateCoordinator
+    ) -> None:
+        """Init method."""
+        super().__init__(device_url, coordinator)
+
+        self._attr_supported_features = (
+            ClimateEntityFeature.TARGET_TEMPERATURE
+            | ClimateEntityFeature.FAN_MODE
+            | ClimateEntityFeature.PRESET_MODE
+        )
+
+        if self.device.states.get(SWING_STATE):
+            self._attr_supported_features |= ClimateEntityFeature.SWING_MODE
+
+        if self._attr_device_info:
+            self._attr_device_info["manufacturer"] = "Hitachi"
+
+    @property
+    def hvac_mode(self) -> HVACMode:
+        """Return hvac operation ie. heat, cool mode."""
+        if (
+            main_op_state := self.device.states[MAIN_OPERATION_STATE]
+        ) and main_op_state.value_as_str:
+            if main_op_state.value_as_str.lower() == OverkizCommandParam.OFF:
+                return HVACMode.OFF
+
+        if (
+            mode_change_state := self.device.states[MODE_CHANGE_STATE]
+        ) and mode_change_state.value_as_str:
+            # The OVP protocol has 'auto cooling' and 'auto heating' values that are equivalent to the HLRRWIFI protocol without spaces
+            sanitized_value = mode_change_state.value_as_str.replace(" ", "").lower()
+            return OVERKIZ_TO_HVAC_MODES[sanitized_value]
+
+        return HVACMode.OFF
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set new target hvac mode."""
+        if hvac_mode == HVACMode.OFF:
+            await self._global_control(main_operation=OverkizCommandParam.OFF)
+        else:
+            await self._global_control(
+                main_operation=OverkizCommandParam.ON,
+                hvac_mode=HVAC_MODES_TO_OVERKIZ[hvac_mode],
+            )
+
+    @property
+    def fan_mode(self) -> str | None:
+        """Return the fan setting."""
+        if (state := self.device.states[FAN_SPEED_STATE]) and state.value_as_str:
+            return OVERKIZ_TO_FAN_MODES[state.value_as_str]
+
+        return None
+
+    @property
+    def fan_modes(self) -> list[str] | None:
+        """Return the list of available fan modes."""
+        return [*FAN_MODES_TO_OVERKIZ]
```

Comment:
Since this never changes, this should be moved to class level:

_attr_fan_modes = [*FAN_MODES_TO_OVERKIZ]